Frameworks/Foundation/NSURLConnection.mm

```diff
@@ -260,37 +294,45 @@ - (void)scheduleInRunLoop:(NSRunLoop*)aRunLoop forMode:(NSRunLoopMode)mode {
             return;
         }
 
-        StrongId<NSRunLoop> retainedLoop = aRunLoop;
-        StrongId<NSString> retainedMode = mode;
-        _scheduledRunLoops.push_back(std::make_pair(retainedLoop, retainedMode));
+        _scheduledRunLoops.emplace_back(aRunLoop, mode);
     }
 }
 
 /**
  @Status Caveat
- @Notes  TODO #: Callbacks do not currently occur on delegate queue
+ @Notes  TODO #2469: Callbacks do not currently occur on delegate queue
 */
 - (void)setDelegateQueue:(NSOperationQueue*)queue {
     @synchronized(self) {
+        if (_started) {
+            TraceError(TAG, L"[NSURLConnection setDelegateQueue:] called on a connection that already started.");
+            return;
+        }
+
         if (_scheduledRunLoops.size() > 0) {
             [NSException raise:NSInternalInconsistencyException
                         format:@"A connection cannot be scheduled with both an operation queue and a run loop."];
         }
 
+        if (_delegateQueue) {
+            TraceError(TAG, L"[NSURLConnection setDelegateQueue:] called on a connection that already had a delegate queue; ignoring.");
+            return;
+        }
+
         _delegateQueue = queue;
     }
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes  TODO #2469: Cannot currently schedule on more than one runloop
 */
 - (void)unscheduleFromRunLoop:(NSRunLoop*)aRunLoop forMode:(NSRunLoopMode)mode {
     @synchronized(self) {
-        for (auto it = _scheduledRunLoops.begin(); it != _scheduledRunLoops.end(); ++it) {
-            if ([it->first isEqual:aRunLoop] && [it->second isEqual:mode]) {
-                _scheduledRunLoops.erase(it);
-                return;
-            }
+        auto it = std::find(_scheduledRunLoops.begin(), _scheduledRunLoops.end());
```

Comment:
this appears to be finding "the run loop"? It doesn't care about its arguments. If that's the case, why not just _scheduledRunLoops.clear()?